fission/src/ui/panels/configuring/scoring/ZoneConfigPanel.tsx

```diff
@@ -1,84 +1,202 @@
+
+    const field = useMemo(() => {
+        const assemblies = [...World.SceneRenderer.sceneObjects.values()]
+        for (let i = 0; i < assemblies.length; i++) {
+            const assembly = assemblies[i]
+            if (!(assembly instanceof MirabufSceneObject)) continue
+
+            if ((assembly as MirabufSceneObject).miraType != MiraType.FIELD) continue
+
+            return assembly
+        }
+
+        return undefined
+    }, [])
+
+    useEffect(() => {
+        closePanel("scoring-zones")
+        configureGizmo()
+    }, [])
+
+    const configureGizmo = () => {
+        transformGizmoRef.current = new TransformGizmos(
+            new THREE.Mesh(
+                new THREE.BoxGeometry(1, 1, 1),
+                World.SceneRenderer.CreateToonMaterial(ReactRgbaColor_ThreeColor(theme.HighlightHover.color))
+            )
+        )
+        transformGizmoRef.current.AddMeshToScene()
+
+        transformGizmoRef.current.CreateGizmo("translate", 1.5)
+
+        const position = SelectedZone.zone.position
+        const rotation = SelectedZone.zone.rotation
+        const scale = SelectedZone.zone.scale
+
+        transformGizmoRef.current.mesh.position.set(position[0], position[1], position[2])
+        transformGizmoRef.current.mesh.rotation.setFromQuaternion(
+            new THREE.Quaternion(rotation[0], rotation[1], rotation[2], rotation[3])
+        )
+        transformGizmoRef.current.mesh.scale.set(scale[0], scale[1], scale[2])
+    }
+
+    const saveSettings = () => {
+        SelectedZone.zone.name = name
+        SelectedZone.zone.alliance = alliance
+        SelectedZone.zone.parentNode = selectedNode
+        SelectedZone.zone.points = points
+        SelectedZone.zone.destroyGamepiece = destroy
+        SelectedZone.zone.persistentPoints = persistent
+
+        if (transformGizmoRef.current != undefined) {
+            const position = transformGizmoRef.current.mesh.position
+            const rotation = transformGizmoRef.current.mesh.quaternion
+            const scale = transformGizmoRef.current.mesh.scale
+
+            SelectedZone.zone.position = [position.x, position.y, position.z]
+            SelectedZone.zone.rotation = [rotation.x, rotation.y, rotation.z, rotation.w]
+            SelectedZone.zone.scale = [scale.x, scale.y, scale.z]
+        }
+
+        PreferencesSystem.savePreferences()
+    }
+
+    /** Sets the selected node if it is a part of the currently loaded field */
+    const trySetSelectedNode = useCallback(
+        (body: Jolt.BodyID) => {
+            if (SelectedZone.zone == undefined || Object.keys(PreferencesSystem.getAllFieldPreferences()).length == 0)
+                return false
+
+            const assoc = World.PhysicsSystem.GetBodyAssociation<RigidNodeAssociate>(body)
+            if (!assoc || assoc?.sceneObject != field) {
+                return false
+            }
+
+            setSelectedNode(assoc.rigidNodeId)
+            return true
+        },
+        [field]
+    )
```

Comment:
Pretty much just copy and paste what I have in intake and ejector. There are a lot of things that have changed and it'll be easier to do it like that.